planning/obstacle_cruise_planner/src/node.cpp

```diff
@@ -56,18 +56,12 @@ std::optional<T> getObstacleFromUuid(
 
 bool isFrontObstacle(
   const std::vector<TrajectoryPoint> & traj_points, const size_t ego_idx,
-  const geometry_msgs::msg::Point & obstacle_pos)
+  const geometry_msgs::msg::Point & obstacle_pos, double * ego_to_obstacle_distance)
 {
   const size_t obstacle_idx = motion_utils::findNearestIndex(traj_points, obstacle_pos);
 
-  const double ego_to_obstacle_distance =
-    motion_utils::calcSignedArcLength(traj_points, ego_idx, obstacle_idx);
-
-  if (ego_to_obstacle_distance < 0) {
-    return false;
-  }
-
-  return true;
+  *ego_to_obstacle_distance = motion_utils::calcSignedArcLength(traj_points, ego_idx, obstacle_idx);
+  return *ego_to_obstacle_distance >= 0.0;
 }
```

Comment:
The reference argument is not recommended.
How about renaming the function to `calcDistanceToFrontVehicle` or something whose return value's type is `std::optional<double>` and return std::nullopt if the vehicle is not a front vehicle?

Reply:
Done, thanks for the suggestion!